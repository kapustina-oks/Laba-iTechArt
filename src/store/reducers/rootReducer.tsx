import { combineReducers } from "redux";
import authReducer from "@/store/reducers/authReducer";
import modalReducer from "@/store/reducers/modalReducer";
import userNameReducer from "@/store/reducers/userNameReducer";
import adminReducer from "@/store/reducers/adminReducer";
import editModalReducer from "@/store/reducers/editModalReducer";
import fetchReducer from "@/store/reducers/fetchReducer";
import cartReducer from "@/store/reducers/cartReducer";

const rootReducer = combineReducers({
  auth: authReducer,
  modal: modalReducer,
  userName: userNameReducer,
  admin: adminReducer,
  editModal: editModalReducer,
  reqGame: fetchReducer,
  cart: cartReducer,
});

export type RootState = ReturnType<typeof rootReducer>;
export default rootReducer;
