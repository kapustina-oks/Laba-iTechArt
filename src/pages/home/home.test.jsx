import { render } from "@testing-library/react";
import Home from "./home";
import "@testing-library/jest-dom";
import { Provider } from "react-redux";
import { configureStore } from "@reduxjs/toolkit";
import rootReducer from "../../store/reducers/rootReducer";
import dataGames from "../../mock/dataBase";
import userEvent from "@testing-library/user-event";

export const { expect } = window;

jest.mock("react-router-dom", () => ({
  useHistory: jest.fn(),
  useLocation: jest.fn().mockReturnValue({
    pathname: "/",
    search: "",
    hash: "",
    state: null,
  }),
}));

const store = configureStore({ reducer: rootReducer });

describe("testing of home page", () => {
  it("should render home page", () => {
    render(
      <Provider store={store}>
        <Home />
      </Provider>
    );
  });

  it("Home snapshot", () => {
    render(<Home />);
    expect(document.body).toMatchSnapshot();
  });

  it("fetches games from an API", () => {
    const mockJSONPromise = Promise.resolve({
      data: dataGames,
    });
    global.fetch = jest.fn().mockImplementationOnce(() => mockJSONPromise);
    const {getByText, findAllByRole} = render(
      <Provider store={store}>
        <Home />
      </Provider>
    );
    expect(global.fetch).toHaveBeenCalledTimes(1);
    expect(document.body).toMatchSnapshot();

  });
});
