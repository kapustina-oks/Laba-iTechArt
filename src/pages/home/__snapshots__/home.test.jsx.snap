// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing of home page fetches games from an API 1`] = `
<body>
  <div>
    <div
      class="home_container"
    >
      <div
        class="row"
      >
        <form
          action=""
          method="get"
        >
          <input
            placeholder="Search game..."
            type="text"
          />
        </form>
      </div>
      <div
        class="new-games"
      >
        Categories
      </div>
      <div
        class="grid_category"
      />
      <div
        class="new-games"
      >
        New games
      </div>
      <div
        class="grid_games"
      />
    </div>
  </div>
</body>
`;
